classes/submission/SubmissionDAO.inc.php

```diff
@@ -471,20 +471,33 @@ function getByUserId($userId, $contextId = null) {
 	 *  whose section will be included in the results (excluding others).
 	 * @param $includeDeclined boolean optional include submissions which have STATUS_DECLINED
 	 * @param $includePublished boolean optional include submissions which are published
+	 * @param $submissionId int|null optional Filter by submission id.
 	 * @param $title string|null optional Filter by title.
 	 * @param $author string|null optional Filter by author.
 	 * @param $stageId int|null optional Filter by stage id.
 	 * @param $sectionId int|null optional Filter by section id.
 	 * @param $rangeInfo DBRangeInfo
 	 * @return DAOResultFactory containing matching Submissions
 	 */
-	function getBySubEditorId($contextId, $subEditorId = null, $includeDeclined = true, $includePublished = true, $title = null, $author = null, $stageId = null, $sectionId = null, $rangeInfo = null) {
+	function getBySubEditorId($contextId, $subEditorId = null, $includeDeclined = true, $includePublished = true, $submissionId = null, $title = null, $author = null, $stageId = null, $sectionId = null, $rangeInfo = null) {
 		$params = $this->getFetchParameters();
 		if ($subEditorId) $params[] = (int) $subEditorId;
 		$params[] = (int) $contextId;
 		$params[] = (int) ROLE_ID_MANAGER;
 		$params[] = (int) ROLE_ID_SUB_EDITOR;
 
+		if ($submissionId) {
```

Comment:
We still have to put the `$params[] = (int) $submissionId;` line of code, right?
The `filter_var($submissionId, FILTER_VALIDATE_INT)` returns false if it does not pass the check. 
We could do something like
 `($submissionId!==null && !$submissionId?' AND s.submission_id = ?':'')` 
but that way, if the filter_var check fails, all the rows will be returned by the query (thus the 
`$params[] = -1;` line).
Am I missing something there?